scripts/scil_denoise_sf.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Script to compute per-vertices hemisphere-aware (asymmetric) Gaussian
+filtering of spherical functions (SF) given an array of spherical harmonics
+(SH) coefficients.
+
+The resulting SF can be expressed using a full SH basis (to keep the
+asymmetry resulting from the filtering) or a symmetric SH basis (where the
+effect of the filtering is a simple denoising).
+"""
+
+import argparse
+import logging
+import os
+
+import nibabel as nib
+import numpy as np
+
+from scilpy.io.utils import (add_overwrite_arg, assert_inputs_exist,
+                             assert_outputs_exist, add_sh_basis_args)
+
+from scilpy.denoise.asym_enhancement import local_asym_gaussian_filtering
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(
+        description=__doc__, formatter_class=argparse.RawTextHelpFormatter)
+
+    p.add_argument('in_sh',
+                   help='Path to the input file.')
+
+    p.add_argument('--out_dir', default='.',
+                   help='Output directory. Default is current directory.')
+
+    p.add_argument('--out_sh', default='out_filtered.nii.gz',
```

Comment:
I would actually prefer if the out_sh was mandatory (leave out_mask optional). --out_dir is still useful like it is, keep it.